formver.compiler-plugin/core/src/org/jetbrains/kotlin/formver/core/embeddings/callables/NonInlineNamedFunction.kt

```diff
@@ -7,21 +7,34 @@ package org.jetbrains.kotlin.formver.core.embeddings.callables
 
 import org.jetbrains.kotlin.formver.core.conversion.StmtConversionContext
 import org.jetbrains.kotlin.formver.core.embeddings.expression.ExpEmbedding
+import org.jetbrains.kotlin.formver.core.embeddings.expression.FunctionCall
 import org.jetbrains.kotlin.formver.core.embeddings.expression.MethodCall
 import org.jetbrains.kotlin.formver.core.embeddings.expression.PlaceholderVariableEmbedding
 import org.jetbrains.kotlin.formver.core.names.PlaceholderReturnVariableName
+import org.jetbrains.kotlin.formver.viper.ast.Function
 import org.jetbrains.kotlin.formver.viper.ast.Method
 
 class NonInlineNamedFunction(val signature: FullNamedFunctionSignature) : RichCallableEmbedding,
     FullNamedFunctionSignature by signature {
     override fun insertCall(
         args: List<ExpEmbedding>,
         ctx: StmtConversionContext,
-    ): ExpEmbedding = MethodCall(signature, args)
+    ): ExpEmbedding {
+        return if (ctx.isPureFunction(signature.symbol)) {
```

Comment:
I think we should already be aware of this earlier, probably when we create this `NonInlineNamedFunction`, or perhaps even when we create the `signature`